src/ucp/core/ucp_ep.c

```diff
@@ -74,6 +74,17 @@ typedef struct ucp_ep_discard_lanes_arg {
 } ucp_ep_discard_lanes_arg_t;
 
 
+/**
+ * Failed UCT EP placeholder
+ */
+typedef struct {
+    uct_ep_t     super;
+    ucs_status_t status;
+    unsigned     counter;
```

Comment:
maybe use uct completion? it already has counter + status and has suitable API, static instance could has empty completion func, and `free(arg)` for EP allocated instance.

Reply:
but seems there is no completion callback needed here?

Reply:
we can invoke it manually like we do it in ucp_flush, for example.
this is not a critical to me, just an idea how to reuse existing code for similar functionality:
- it has a counter
- it stores a first happening error status
- it has a function pointer which can be defined depends on 2 different cases